src/main/java/com/aws/greengrass/deployment/DeploymentService.java

```diff
@@ -402,19 +418,36 @@ private void finishCurrentDeployment() throws InterruptedException {
     }
 
     private void persistGroupToRootComponents(DeploymentDocument deploymentDocument) {
-        Map<String, Object> deploymentGroupToRootPackages = new HashMap<>();
         Topics deploymentGroupTopics = config.lookupTopics(GROUP_TO_ROOT_COMPONENTS_TOPICS);
         Topics groupMembershipTopics = config.lookupTopics(GROUP_MEMBERSHIP_TOPICS);
         deploymentGroupTopics.forEach(node -> {
-            Topics groupTopics = (Topics) node;
-            if (groupMembershipTopics.find(groupTopics.getName()) == null
-                    && !groupTopics.getName().startsWith(DEVICE_DEPLOYMENT_GROUP_NAME_PREFIX)
-                    && !groupTopics.getName().equals(LOCAL_DEPLOYMENT_GROUP_NAME)) {
-                logger.info("Removing mapping for thing group " + groupTopics.getName());
-                groupTopics.remove();
+            if (node instanceof Topics) {
+                Topics groupTopics = (Topics) node;
+                if (groupMembershipTopics.find(groupTopics.getName()) == null && !groupTopics.getName()
+                        .startsWith(DEVICE_DEPLOYMENT_GROUP_NAME_PREFIX) && !groupTopics.getName()
+                        .equals(LOCAL_DEPLOYMENT_GROUP_NAME)) {
+                    logger.info("Removing mapping for thing group " + groupTopics.getName());
+                    groupTopics.remove();
+                }
+            }
+        });
+
+        Topics groupLastDeploymentTopics = config.lookupTopics(GROUP_TO_LAST_DEPLOYMENT_TOPICS);
+        groupLastDeploymentTopics.forEach(node -> {
+            if (node instanceof Topics) {
+                Topics groupTopics = (Topics) node;
+                if (groupMembershipTopics.find(groupTopics.getName()) == null && !groupTopics.getName()
+                        .startsWith(DEVICE_DEPLOYMENT_GROUP_NAME_PREFIX) && !groupTopics.getName()
+                        .equals(LOCAL_DEPLOYMENT_GROUP_NAME)) {
+                    logger.info("Removing last deployment information for thing group " + groupTopics.getName());
```

Comment:
I don't think this needs to be info.

Reply:
At what level this should be logged then?

Reply:
No, lesser than that. Debug